lib/utils.js

```diff
@@ -304,4 +310,32 @@ utils.mapHeaders = function mapHeaders(headers, addHeaders) {
     return headers;
 };
 
+utils.isUnsupportedQuery = function isUnsupportedQuery(queryStr) {
+    if (!queryStr) {
+        return false;
+    }
+    // Reject unsupported query fields (e.g. ?accelerate), but allow
+    // the same strings as values (e.g. ?test=accelerate) and queries like
+    // '?acceleratefoobar' or '&foobaraccelerate'
+    return constants.unsupportedQueries.some(q => {
```

Comment:
I don't mind running a few performance tests to get some experience with that since I haven't run performance tests before. As long as merging this PR is not urgent, which I don't think it is at the moment.